src/shared/requests.ts

```diff
@@ -0,0 +1,91 @@
+/*****
+ License
+ --------------
+ Copyright © 2020 Mojaloop Foundation
+ The Mojaloop files are made available by the Mojaloop Foundation under the
+ Apache License, Version 2.0 (the 'License') and you may not use these files
+ except in compliance with the License. You may obtain a copy of the License at
+ http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, the Mojaloop files
+ are distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+ Contributors
+ --------------
+ This is the official list of the Mojaloop project contributors for this file.
+ Names of the original copyright holders (individuals or organizations)
+ should be listed with a '*' in the first column. People who have
+ contributed from an organization can be listed under the organization
+ that actually holds the copyright for their contributions (see the
+ Gates Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+
+ - Abhimanyu Kapur <abhi.kapur09@gmail.com>
+
+ --------------
+ ******/
+import { putConsents } from '@mojaloop/sdk-standard-components'
+import { Consent } from '../model/consent'
+import { Scope } from '../model/scope'
+import { scopeDb } from '../lib/db'
+import { Enum } from '@mojaloop/central-services-shared'
+
+/**
+ * Builds body of outgoing request and makes PUT consents/{ID} call to server
+ * @param consent Consent object with credential challenge, type and status
+ * @param headers headers from PISP generate challenge request
+ */
+
+// TODO: Figure out return type of putConsents() and assign type accordingly
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export async function putConsentId (consent: Consent, headers): Promise<any> {
+  // Switch SOURCE and DESTINATION in headers
+  const destinationId = headers[Enum.Http.Headers.FSPIOP.SOURCE]
+  // eslint-disable-next-line max-len
+  headers[Enum.Http.Headers.FSPIOP.SOURCE] = headers[Enum.Http.Headers.FSPIOP.DESTINATION]
+  headers[Enum.Http.Headers.FSPIOP.DESTINATION] = destinationId
+
+  // Retrieve scopes
+  const scopesRetrieved: Scope[] = await scopeDb.retrieve(consent.id)
+
+  // Reformat scopes to match what external handler wants
```

Comment:
This is taking a list of actions and putting it into an array attached to the consent.